### PR TITLE
feat: weather endpoint

### PR DESCRIPTION
**Pull request by vitormcferreira**
## Describe your changes
Add a endpoint to generate random weather.

## Link to issue this resolves
Closes [#296](https://github.com/ageddesi/Mocked-API/issues/296).

## Screenshot of changes(if relevant)
image

### CODE REVIEW

File: src/modules/weather/tests/api/weather-routes.test.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const request = require('supertest');
+import app from '../../../../../app';
+
+describe('vehicles api endpoints', () => {
```
Comment (ageddesi) [inline src/modules/weather/tests/api/weather-routes.test.ts:4]: Hi, just a minor spelling mistake here.

File: src/modules/weather/utils/getWeather.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    } else if (temperatureC <= 35) {
+        return 'Warm';
+    } else if (temperatureC <= 40) {
+        return 'Hot';
+    } else {
+        return 'Scorching';
+    }
+};
+
+const toFahrenheit = (temperatureC: number) => Number((temperatureC * (9 / 5) + 32).toFixed(1));
```
Comment (ageddesi) [inline src/modules/weather/utils/getWeather.ts:29]: please pull toFahrenheit into its own function in utils folder please.

File: src/modules/weather/utils/getWeather.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+// temperature in Celsius from https://en.wikipedia.org/wiki/Earth
+const randomCelsius = () => faker.datatype.float({ min: -89.2, max: 56.7, precision: 0.1 });
```
Comment (ageddesi) [inline src/modules/weather/utils/getWeather.ts:32]: please pull randomCelsius into its own function in utils folder please.

File: src/modules/weather/api/weather-routes.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+     *     responses:
+     *       '200':
+     *         description: Obtain an object representing random data about weather
+     *         schema:
+     *           $ref: '#/definitions/MockWeather'
+     */
+    app.get('/weather', (req: Request, res: Response) => {
+        res.json(getWeather());
+    });
+};
```
Comment (ageddesi) [inline src/modules/weather/api/weather-routes.ts:22]: Are you able to add a second endpoint that takes in a qty please. /weather/:qty
you can use our util function getQtyFromRequest to help with this.

File: src/modules/weather/api/weather-routes.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import * as core from 'express-serve-static-core';
+import { getWeather } from '../utils/getWeather';
+
+module.exports = function (app: core.Express) {
+    /**
+     * @openapi
+     * '/weather/':
+     *   get:
+     *     tags:
+     *     - Weather
```
Comment (ageddesi) [inline src/modules/weather/api/weather-routes.ts:11]: Can you please also add this tag to the utils/swagger.ts file please.

**Review by ageddesi — CHANGES_REQUESTED** [review]
Amazing work, just a few minor changes and will be ready to be merged

**Comment by ageddesi** [comment]
Hi @vitormcferreira Are you able to make the changes outstanding on this PR?

File: src/modules/weather/tests/api/weather-routes.test.ts (hunk 1 above)
Comment (vitormcferreira) [reply src/modules/weather/tests/api/weather-routes.test.ts]: Adjusted.

File: src/modules/weather/utils/getWeather.ts (hunk 2 above)
Comment (vitormcferreira) [reply src/modules/weather/utils/getWeather.ts]: Adjusted.

File: src/modules/weather/utils/getWeather.ts (hunk 3 above)
Comment (vitormcferreira) [reply src/modules/weather/utils/getWeather.ts]: Adjusted.

File: src/modules/weather/api/weather-routes.ts (hunk 5 above)
Comment (vitormcferreira) [reply src/modules/weather/api/weather-routes.ts]: Adjusted.

**Comment by vitormcferreira** [comment]
Sorry for the delay in corrections.

I made the requested corrections and some code improvements.

**Comment by ageddesi** [comment]
Thank you @vitormcferreira for the changes and contributing to the project :) 👍

**Pull request merged**